djmoney/contrib/django_rest_framework/fields.py

```diff
@@ -7,6 +10,10 @@
 from djmoney.models.validators import MaxMoneyValidator, MinMoneyValidator
 from djmoney.money import Money
 from djmoney.utils import MONEY_CLASSES, get_currency_field_name
+from moneyed.classes import CurrencyDoesNotExist
+
+
+PrimitiveMoney = namedtuple("PrimitiveMoney", ["amount", "currency"])
```

Comment:
What is the primary purpose of this class and why `Money` won't work instead?

Reply:
Oh, I think that it is here to avoid validation inside `Money`, right, so it is deferred to `to_internal_value`?

Reply:
Exactly, creation of a Money instance and validation of its data will now only happen in to_internal_value. PrimitiveMoney will only "transport" the data to to_internal_value without doing any validation

Reply:
Cool! I'd suggest replacing `namedtuple` with a handwritten class with `__slots__` (as it is a bit faster and potentially harder to misuse) and renaming it to `_PrimitiveMoney` to denote, its private use. What do you think?

Reply:
Also, a short comment describing the purpose of this class would be helpful

Reply:
> Cool! I'd suggest replacing `namedtuple` with a handwritten class with `__slots__` (as it is a bit faster and potentially harder to misuse)

The only (small) drawback I see with only `__slots__` is immutability, which I kind of like in cases like this. Though I won't say I have a very strong opinion about it here. It seems like it's possible to combine `namedtuple` and `__slots__` to get an immutable  object. What do you think, should we do that?

> renaming it to `_PrimitiveMoney` to denote, its private use

Sounds good!

> Also, a short comment describing the purpose of this class would be helpful

Definitely, I'll add in a comment

Reply:
@Stranger6667 I bumped to use the `namedtuple` + `__slots__` combination just to get a feel of what that would look like. Let me know if you want me to change it.

Reply:
Maybe something like this to avoid using `namedtuple` at all?

```python
class _PrimitiveMoney:
    __slots__ = ("amount", "currency")

    def __init__(self, amount, currency):
        self.amount = amount
        self.currency = currency
```

Reply:
Alright, I bumped to your suggestion.